src/ng-select/ng-select.component.ts

```diff
@@ -489,14 +489,19 @@ export class NgSelectComponent implements OnDestroy, OnChanges, AfterViewInit, C
         }
 
         const select = (val: any) => {
-            const item = this.itemsList.findItem(val);
+            let item = this.itemsList.findItem(val);
             if (item) {
                 this.itemsList.select(item);
             } else {
                 const isObject = val instanceof Object;
                 const simpleValue = !isObject && !this.bindValue;
                 if (isObject || simpleValue) {
                     this.itemsList.select(this.itemsList.mapItem(val, simpleValue, null));
+                } else if (!isObject && this.bindValue && this._isTypeahead) {
```

Comment:
https://github.com/ng-select/ng-select/pull/312/files#diff-504437db9e7ef93a33fdb3cad468d194R418
what about this line? is it really necessary?

Reply:
This won't work. Imagine you have id as `bindValue`, everytime you look up new items, labels will switch to id's 👹

Reply:
That is correct, but in this case when typeahead is used with bindValue you don't have any other choose but only show bindValue as label, so user is responsible for it and he can show something that make sense, city name, user name, email ....

Reply:
But it shouldn't happen for correctly selected items. When you select you have normal label, and after lookup it switches to id..

Reply:
True, solution by not removing selected items sounds better 👍